### ISSUE TITLE
`_` in `\u` char escapes is wrongly marked as an error

### ISSUE THREAD

**Issue by CAD97** [intellij-rust/intellij-rust#2353]
```rust
fn main() {
    println!("{}", "\u{1_F602}");
    println!("{}", '\u{1_F602}');
}
```

![2018-03-07](https://user-images.githubusercontent.com/5992217/37077946-1d76353a-21ab-11e8-8f54-8435a9306be7.png)

**Comment by Undin** [comment]
Fixed in #2405
